razers and splazers skip files? [former: clang5 warnings]

http://cdash.seqan.de/viewBuildError.php?type=1&buildid=246555

More warnings, yay!

We have the following warning:

```terminal
seqan-src/apps/razers/outputFormat.h:1204:7: warning: variable 'filecount' is incremented both in the loop header and in the loop body [-Wfor-loop-analysis]
                                ++filecount;
                                  ^
```

The offending code is the following:
```c++
for (unsigned filecount = 0; filecount < length(genomeFileNameList); ++filecount)
{
    // open genome file
    SeqFileIn gFile;
    if (!open(gFile, toCString(genomeFileNameList[filecount])))
    {
        std::cerr << "Couldn't open genome file." << std::endl;
        break;
    }
   // more code...
    close(gFile);
    ++filecount;
}
```

Razers and splazers have this structure and therefore the warning:
https://github.com/seqan/seqan/blob/a0691b9d05eb1ee56f513a5642f745231d77798b/apps/razers/outputFormat.h#L936-L1205
https://github.com/seqan/seqan/blob/a0691b9d05eb1ee56f513a5642f745231d77798b/apps/splazers/outputFormat.h#L1114-L1362

Even razers has this line, but it is commented out.
https://github.com/seqan/seqan/blob/a0691b9d05eb1ee56f513a5642f745231d77798b/apps/razers3/outputFormat.h#L982-L1162

The issue dates back to the svn import and it is not clear why the code does effectively `filecount+=2`. We have no test-cases that test this option (gff with multiple input genome sequences). This case only happens, if the given genome file is not a fasta file, but contains a list of file names, i.e.:

```
/home/user/mypath/file1.fasta
/home/user/mypath/file2.fasta
/home/user/mypath/file3.fasta
```

Some other code places suggest that it should be `filecount++`:
https://github.com/seqan/seqan/blob/a0691b9d05eb1ee56f513a5642f745231d77798b/apps/razers/razers.h#L1909-L2068

@h-2, @rrahn I need some feedback

--- a/apps/razers/outputFormat.h
+++ b/apps/razers/outputFormat.h
@@ -934,7 +934,12 @@ void dumpMatches(
 			}
 			break;
 		case 3: // Gff:  printf "$chr $name_$format read $pos %ld . $dir . ID=$col[0]$unique$rest\n",$pos+$len-1;
-			for (unsigned filecount = 0; filecount < length(genomeFileNameList); ++filecount)
+			// NOTE(marehr): filecount+=2 might be a potential bug [https://github.com/seqan/seqan/issues/2165]
+			// In revision 4dbf27b55 and before, filecount was incremented twice at the
+			// end of the for loop, which caused a compiler warning (once in the body
+			// and once in the iteration_expression of the for loop). We kept this
+			// behaviour, because we have no active maintainer for this app.
+			for (unsigned filecount = 0; filecount < length(genomeFileNameList); filecount+=2)
 			{
 				// open genome file	
 				SeqFileIn gFile;
@@ -1201,7 +1206,6 @@ void dumpMatches(
 					}
 				}
 				close(gFile);
-				++filecount;
 			}
 			break;
 	}

--- a/apps/splazers/outputFormat.h
+++ b/apps/splazers/outputFormat.h
@@ -1112,7 +1112,12 @@ void dumpMatches(
 			}
 			break;
 		case 3: // Gff:  printf "$chr $name_$format read $pos %ld . $dir . ID=$col[0]$unique$rest\n",$pos+$len-1;
-			for (unsigned filecount = 0; filecount < length(genomeFileNameList); ++filecount)
+			// NOTE(marehr): filecount+=2 might be a potential bug [https://github.com/seqan/seqan/issues/2165]
+			// In revision 4dbf27b55 and before, filecount was incremented twice at the
+			// end of the for loop, which caused a compiler warning (once in the body
+			// and once in the iteration_expression of the for loop). We kept this
+			// behaviour, because we have no active maintainer for this app.
+			for (unsigned filecount = 0; filecount < length(genomeFileNameList); filecount+=2)
 			{
 				// open genome file	
 				SeqFileIn gFile;
@@ -1358,7 +1363,6 @@ void dumpMatches(
 					}
 				}
 				close(gFile);
-				++filecount;
 			}
 			break;
 		case 4: // SAM output for splitMatches

